internal/pipeline/pipeline_function.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package pipeline provides struct definitions for Pipeline and utility
+// methods to read and write a pipeline resource.
+package pipeline
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/GoogleContainerTools/kpt/internal/pipeline/runtime"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	builtinSetAnnotatorImage string = "builtin/set-annotator"
```

Comment:
I wouldn't start by having concept of built-ins. Instead just support container-based functions for now. Per the other comment, need to have a clean between the orchestration logic and example functions.

Reply:
Using a builtin function allows us to run pipeline tests without docker.